include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -75,34 +75,60 @@ struct __subgroup_radix_sort
         }
     };
 
-    template <typename _KeyT, typename _Wi, typename _Src, typename _Keys>
+    template <typename _ValT, bool __uniform_space, typename _Wi, typename _Src, typename _Vals>
     static void
-    __block_load(const _Wi __wi, const _Src& __src, _Keys& __keys, const uint32_t __n, const _KeyT& __default_key)
+    __block_load(const _Wi __wi, const _Src& __src, _Vals& __vals, const uint32_t __n)
     {
 #pragma unroll
         for (uint16_t __i = 0; __i < __block_size; ++__i)
         {
-            const uint16_t __offset = __wi * __block_size + __i;
-            if (__offset < __n)
-                __keys[__i] = __src[__offset];
+            const uint16_t __idx = __wi * __block_size + __i;
+            if (__idx < __n)
+                __vals[__i] = __src[__idx];
             else
-                __keys[__i] = __default_key;
+            {
+                if constexpr (__uniform_space)
+                {
+                    //we use numeric_limits::lowest for floating-point types with denormalization,
+                    //due to numeric_limits::min gets the minimum positive normalized value
+                    static constexpr _ValT __default_val =
+                        __is_asc ? std::numeric_limits<_ValT>::max() : std::numeric_limits<_ValT>::lowest();
+
+                    __vals[__i] = __default_val;
+                }
```

Comment:
Does this make sense?  Whether or not `_KeyT` is the same as `_ValueT`,  where `_KeyT` is the type coming out of the projection operator, it doesn't mean that these are valid default (identity) values unless we are skipping applying the projection operator to them.  

What if  `_ValueT` is int, and the projection operator is `[](const int& a) {return -1 * a;}`, and therefore `_KeyT` is also int.  Effectively, the projection operator would reverse `__is_asc` and these default values would fail to clamp their filler values to the end of the list.

Reply:
the essence here that we cannot define a comparison identical value for non scalar arithmetic type... 
BTW, it make sense to add static assert that `__is_radix_sort_usable_for_type<oneapi::dpl::__internal::__key_t<_Proj, _Range>, _Compare>::value` is true.
Regarding `[](const int& a) {return -1 * a;}` I thought about that.... Please have a look an example:
1) __is_asc is true:
3 6  8 3 4 0xfff
sorted: 3 3 4 6 8 0xfff
3 6  8 3 4 0xfff -> -3 -6  -8 -3 -4 0xfff
sorted: (-)8 (-)6 (-)4 (-)3 (-)3 0xfff

2) __is_asc is false:
3 6  8 3 4 -2147483648
sorted: 8 6 4 3 3 -2147483648
3 6  8 3 4 0xfff -> -3 -6  -8 -3 -4 -2147483648
sorted: (-)3 (-)3 (-)4 (-)6 (-)8 -2147483648

Reply:
...sorry, it seems I forgot apply the projection for an default (comparison identical) value.. 
So,
__is_asc is true:
3 6  8 3 4 0xfff
sorted: 3 3 4 6 8 0xfff
3 6  8 3 4 0xfff -> -3 -6  -8 -3 -4 -0xfff
sorted: (-)0xfff (-)8 (-)6 (-)4 (-)3 (-)3 => result; 0xfff  8  6  4 3 -> wrong....

Reply:
So, it looks that we can use the uniform space just in case of an identity projection
`constexpr bool __uniform_space = std::is_same_v<_KeyT, _ValT> && (std::is_same_v,<_Proj, oneapi::dpl::identity> || std::is_same_v,<_Proj, std::identity>)`;

Reply:
Yes, I think that is right.  There is some class of projection operators larger than just the identity which would preserve the sortedness which depends on type, but checking for the identity likely covers the only important case here.

Reply:
this comment is not actual any more due to the code was re-written another way.